src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
+                    {
+                        assm = Assembly.Load(assemblyName);
+                    }
+                    catch { }
+
+                    if (assm == null)
+                    {
+                        try
+                        {
+                            AssemblyName asmName = new AssemblyName(assemblyName);
+                            assm = Assembly.Load(asmName.Name);
+                        }
+                        catch { }
+                    }
+                }
+                else
+                {
+                    try
+                    {
+                        assm = Assembly.Load(new AssemblyName(assemblyName));
+                    }
+                    catch { }
```

Comment:
I'm missing something. In Desktop, if bSimpleAssembly==true, and the load succeeds, it calls `ObjectReader.GetSimplyNamedTypeFromAssembly`. if bSimpleAssembly==false and the load succeeds, it calls FormatterServices.GetTypeFromAssembly. If either load fails it calls neither.

In this code, whatever the value of bSimplAssembly, if either load succeeds it calls GetSimplyNamedTypeFromAssembly otherwise it calls FormatterServices.GetTypeFromAssembly.

What am I missing?